src/utils/permissions.ts

```diff
@@ -191,8 +143,43 @@ export async function hasChannelPermissions(
   if (permissions.every((perm) => allowedPermissions.has(perm))) return true;
 
   // Some permission was not explicitly allowed so we default to checking role perms directly
-  const hasPerms = await botHasPermission(guild.id, permissions);
-  return hasPerms;
+  return hasServerPerm(memberID, guild.id, permissions);
+}
+
+/** Returns you an array of the permissions that are not in totalBits */
+export function missingPermissions(
+  totalBits: string,
+  permissions: Permission[],
+) {
+  const missing: Permission[] = [];
+  permissions.forEach((permission) => {
+    if (!(BigInt(totalBits) & BigInt(Permissions[permission]))) {
+      missing.push(permission);
+    }
+  });
+
+  return missing;
+}
+
+/** Throws an error if the given member has not the permissions in the given guild */
+export async function throwOnMissingServerPermission(
+  memberID: string,
+  guildID: string,
+  permissions: Permission[],
+) {
+  const permissionBits = await calculateServerPerm(memberID, guildID);
+
+  const missing = missingPermissions(permissionBits, permissions);
+  if (missing.length) {
+    throw new Error(Errors[`MISSING_${missing[0]}` as Errors]);
+  }
+}
+
+/** Checks if the permissions matches the permission bits */
+export function validatePerms(perms: string, permissions: Permission[]) {
```

Comment:
Similarly, could you rename it to `validatePermissions`?
Furthermore, is the type of `perms` supposed be string or number?

Reply:
> Furthermore, is the type of perms supposed be string or number?

What do you mean with that?